### ISSUE TITLE
Compiling from source code vernemq and installing it

### ISSUE THREAD

**Issue by HDenied** [vernemq/vernemq#1851]
### Environment

- VerneMQ Version: 1.12.1
- OS: Debian 10
- Erlang/OTP version (if building from source): OTP22
- VerneMQ configuration (vernemq.conf) or the changes from the default
- Cluster size/standalone: standalone

### Expected behaviour

It should build with a version of OTP >-21.2 following the READEME.md instruction. Adding vernemq.service to systemd, should work out of the box if the file .service file points to the right files

### Actual behaviour
I had to compile manually the source code with "make rel" command becaise the stable release .service file wasn't working properly: after installing it I couldn't restart the daemon using systemctl, the program was hanging and not connecting.

What I found out and how I fixed it:
It doesn't built with OTP24 giving back a lot of error, you need to use OTP22, not sure about OTP23. I recompiled Erlang to version OTP22 at system level.I manually installed the .service file which gives me back the same error as I was using the .deb package with the exception that probably now that I recompiled with OPT22, when it hangs I can press CTRL+C and the I got back control of the shell, the process was working. I solved the issue changing the .service file as I suggest below:

What I suggest:
- In the docs put an upper limit to the OTP supported
- In the service file to avoid it to hang change from "Type=FORK" to "Type=Simple", the program won't hang anymore.

- Please include minimal steps to reproduce the issue, if possible.
Just build src code with "make rel" with OTP24. if you build the src code.
If you use the .deb package you will see immediately .service file will have weird behaviour and you can't restart the program properly.

**Comment by ioolkos** [comment]
@HDenied Thanks, yes Verne does not compile yet with OTP 24. There's also a difference in building Verne from source (then it won't be a .deb package or use systemd). 

Installing the .deb package, you might have run into a .deb specific issues mentioned here (see "note"): https://github.com/vernemq/vernemq/releases/tag/1.12.1

**Comment by HDenied** [comment]
I tried a more "hacky way": compiling by myself and then creating the systemd files to make it run and restart automatically. What I did wasn't modifying the .postinst file but creating the vernemq account and then adding the little modification in the .service file related to the type of process (as I mentioned from FORK to SIMPLE), which allows  to restart the application smoothly avoiding the program to hang. It works both ways I believe, thank you anyway!
